src/main/rx/cc2500_redpine.c

```diff
@@ -0,0 +1,194 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+#include <sys/_stdint.h>
+
+#include "platform.h"
+
+#ifdef USE_RX_REDPINE_SPI
+
+#include "build/build_config.h"
+#include "build/debug.h"
+
+#include "pg/rx.h"
+#include "pg/rx_spi.h"
+
+#include "common/maths.h"
+#include "common/utils.h"
+
+#include "config/feature.h"
+
+#include "drivers/adc.h"
+#include "drivers/rx/rx_spi.h"
+#include "drivers/rx/rx_cc2500.h"
+#include "drivers/io.h"
+#include "drivers/io_def.h"
+#include "drivers/io_types.h"
+#include "drivers/resource.h"
+#include "drivers/system.h"
+#include "drivers/time.h"
+
+#include "fc/config.h"
+
+#include "rx/rx_spi_common.h"
+#include "rx/cc2500_common.h"
+#include "rx/cc2500_frsky_common.h"
+#include "rx/cc2500_frsky_shared.h"
+
+#include "cc2500_redpine.h"
+
+extern const uint16_t crcTable[];
+
+
+#define TOTAL_PACKET_TIME 1500
+#define CHANNEL_START 3
+void redpineSetRcData(uint16_t *rcData, const uint8_t *packet)
+{
+    uint16_t channelValue;
+
+    //4 stick channels (11-bit)
+    channelValue = (uint16_t)((packet[CHANNEL_START+1] << 8) & 0x700) | packet[CHANNEL_START];
+    rcData[0] = channelValue * 2.0f / 3 + 860 - 64 * 2 / 3;
```

Comment:
What's the reason behind ```* 2.0f```?
Input and output is uint16_t.
Also using all those magic numbers calculation would be better in a macro.

Reply:
I had pulled in the calc for what orginally done to unpack the channel value from frsky_x.  It looks like that has been changed to a better calculation: `((channelValue - 64) * 2 + 860 * 3) / 3;` So I'll swap that in there and make a macro.   

Or if someone knows of a better stick value to 11-bit number translation that doesn't require the divides (maybe divide by 2 to get the compiler to optimize to bitshifts), I'd be open to that also.

Reply:
This maps to `<817.333, 2182>`

`(2 * channelValue + 1226) / 3` may be best representation (gcc shall be able to replace division with reciprocal fixed point multiplication)

The idea behind this is `1500 + (channelValue - 1024) * (2/3)` - scale by 2/3 with center at 1500. Using this formula has problem with integer division - truncate toward zero will make 1500 interval a bit larger (or interval around 64 in equation above)

Reply:
@ledvinap Yeah I like that a lot better, way less calculations going on.  I think you meant `(2 * channelValue + 2452) / 3)' or `(2 * (channelValue + 1226)) / 3)'

Reply:
Slightly OT: but if rcdata was bumped up 10000-20000 instead of 1000-2000 we'd save much of the resolution (or switch to float), but probably not worth it with all that code that touches...

Yeah, we are basically shoving a 11-bit RC data in to 10ish-bits of rcdata

Reply:
-1.. 1 float interval would make some sense. But aux, throttle, deadband, rates, expo etc will complicate it..